### PR TITLE
Add pronouns to Parent

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-13764
affects: parent
release note: Added support for user pronouns

Test plan:
* Log in to Parent > narmstrong > o
* Pronouns should show for Student 1 in the following places:
  - Dashboard: Student Picker and Nav title
  - Manage Children > Children list
  - Manage Children > tap Student1 > Nav title
  - Inbox > Tap Conversation ("Test Push Notification") > to and from labels
  - Inbox > Compose > Course picker > Student name below course
  - Compose > Edit recipients > Should show in the picker > Select Student 1 > Should show in the recipient pills

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1020746&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
#### Release Note: 
Added support for user pronouns

#### Affected Apps: Parent

[MBL-13764](https://instructure.atlassian.net/browse/MBL-13764)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.52% | 93.51% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1020747&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1020768&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Parent/Parent/Dashboard/DashboardViewController.swift [hunk 1]
```diff
@@ -73,7 +73,8 @@ class DashboardViewController: UIViewController, CustomNavbarProtocol {
                 currentStudentID = student.id
                 let color = ColorScheme.observee(student.id).color
                 alertsTabItem.badgeColor = color
-                navbarNameButton.setTitle(student.name, for: .normal)
+                let displayName = Core.User.displayName(student.name, pronouns: student.pronouns)
+                navbarNameButton.setTitle(displayName, for: .normal)
```
Comment (shichongrui) [inline Parent/Parent/Dashboard/DashboardViewController.swift]: Do we need to add these to the calendar and alerts screen as well?

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1020772&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Parent/Parent/Dashboard/DashboardViewController.swift (hunk 1 above)
Comment (armstrongnate) [reply Parent/Parent/Dashboard/DashboardViewController.swift]: No, looks like this takes care of the other tabs.

**Pull request merged**